File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    day = []
    day_sum = 0
    count = 0
    for i in range(len(progresses)):    
        for j in range(progresses[i],100,speeds[i]):
            count+=1
        day.append(count)
        count=0
    
    print(day)
    day_count = 1
    pro_f = day[0]
    for i in range(1,len(day)):
        
            
        if pro_f < day[i]:
            answer.append(day_count)
            pro_f = day[i]
            day_count = 1
        else:
            day_count +=1
        if i == len(day)-1:
            answer.append(day_count)
        
        
    
        
    return answer

File: 프로그래머스/lv2/42586. 기능개발/기능개발.java
import java.util.ArrayList;
import java.util.List;


class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] answer = new int[]{};
        List<Integer> answerData = new ArrayList<>();
        List<Integer> dayCount = new ArrayList<>();

        for(int i=0; i< progresses.length; i++){
            int count = 0;
            for(int j=progresses[i]; j<100; j+=speeds[i]){
                count+= 1;
            }
            dayCount.add(count);
        }

        int standard = dayCount.get(0);
        int count = 1;
        for(int i=1; i< dayCount.size(); i++){
            if( standard < dayCount.get(i)){
                answerData.add(count);
                standard = dayCount.get(i);
                count = 1;
            }
            else{
                count ++;
            }

            if( i == dayCount.size()-1){
                answerData.add(count);
            }

        }
        System.out.println(answerData);
        answer = answerData.stream()
                .mapToInt(i ->i)
                .toArray();
        return answer;
    }
}